pkg/logging/constants.go

```diff
@@ -65,3 +68,12 @@ func From(obj metav1.Object) []interface{} {
 		obj.GetName(),
 	}
 }
+
+// Key is like cache.MetaNamespaceKeyFunc, but with a restricted input set, so it can't error.
```

Comment:
I'm open to doing something where you pass in a prefix and get e.g. `{"foo.workspace": "root", "foo.namespace": "ns", "foo.name": "something"}` as well, not entirely clear to me which is better.

Reply:
would prefer `From` and `Key` to give the same keys, wdyt?

Reply:
klog and go-logr onl support strings? `foo: {workspace: root, ns: bla name: blub}` would really be the natural representation.

Reply:
there is another issue with either of these approaches: I would argue the main usecase is to filter logs by "show me everything that involves  APIExport `foo|bar`". Can a structured query e.g. in Grafana/Loki filter like that? With the key this maybe the easiest, because a non-structured query just by text could find it.

Reply:
This is for ancillary objects, not the particular object being queued by the controller. Yes, generally query languages allow for some loose string matching on keys.